src/test/java/com/google/api/generator/engine/ast/JavaDocCommentTest.java

```diff
@@ -62,12 +62,65 @@ public void createJavaDocComment_sampleCode() {
         JavaDocComment.builder().addComment(comment).addSampleCode(sampleCode).build();
     String expected =
         "sample codes:\n"
-            + "<pre><code>\n"
+            + "<pre>{@code\n"
             + "resource = project/{project}/shelfId/{shelfId}\n"
-            + "</code></pre>";
+            + "}</pre>";
     assertEquals(javaDocComment.comment(), expected);
   }
 
+  @Test
+  public void createJavaDocComment_sampleCodePreserveIndentAndLineBreaks() {
+    String comment = "sample codes:";
+    String sampleCode1 =
+        "SubscriptionAdminSettings subscriptionAdminSettings =\n"
+            + "    SubscriptionAdminSettings.newBuilder().setEndpoint(myEndpoint).build();\n";
+    String sampleCode2 =
+        "SubscriptionAdminSettings subscriptionAdminSettings =\n"
+            + "    SubscriptionAdminSettings\n"
+            + "        .newBuilder()\n"
+            + "        .setEndpoint(myEndpoint)\n"
+            + "        .build();\n";
+    String sampleCodeNotFormatted =
+        "SubscriptionAdminSettings subscriptionAdminSettings =\n"
+            + "    SubscriptionAdminSettings\n"
+            + "        .newBuilder()\n"
+            + "    .setEndpoint(myEndpoint)\n"
+            + "        .build();\n";
+    JavaDocComment javaDocComment1 =
+        JavaDocComment.builder().addComment(comment).addSampleCode(sampleCode1).build();
+    JavaDocComment javaDocComment2 =
+        JavaDocComment.builder().addComment(comment).addSampleCode(sampleCode2).build();
+    JavaDocComment javaDocComment3 =
+        JavaDocComment.builder().addComment(comment).addSampleCode(sampleCodeNotFormatted).build();
+    String expected1 =
```

Comment:
Can the expected values just concatenate the sample code string variables above?